src/serac/numerics/functional/detail/hexahedron_Hcurl.inl

```diff
@@ -442,41 +421,45 @@ struct finite_element<Geometry::Hexahedron, Hcurl<p>> {
       }
     }
 
-    cache_type_tmp<q> cache{};
-
     // to clarify which contractions correspond to which spatial dimensions
     constexpr int x = 2, y = 1, z = 0;
 
     // clang-format off
     //  r(0, dz, dy, dx) = s(0, qz, qy, qx) * B2(qz, dz) * B2(qy, dy) * B1(qx, dx)
     //                   + f(1, qz, qy, qx) * G2(qz, dz) * B2(qy, dy) * B1(qx, dx)
     //                   - f(2, qz, qy, qx) * B2(qz, dz) * G2(qy, dy) * B1(qx, dx);
-    cache.A2.x[0] = contract< z, 0 >(source[0], B2) + contract< z, 0 >(flux[1], G2);
-    cache.A2.x[1] = contract< z, 0 >(flux[2], B2);
-    cache.A1.x = contract< y, 0 >(cache.A2.x[0], B2) - contract< y, 0 >(cache.A2.x[1], G2);
-    element_residual[0].x += contract< x, 0 >(cache.A1.x, B1);
+    {
+      auto A20 = contract< z, 0 >(source[0], B2) + contract< z, 0 >(flux[1], G2);
+      auto A21 = contract< z, 0 >(flux[2], B2);
+      auto A1 = contract< y, 0 >(A20, B2) - contract< y, 0 >(A21, G2);
+      element_residual[0].x += contract< x, 0 >(A1, B1);
+    }
 
     //  r(1, dz, dy, dx) = s(1, qz, qy, qx) * B2(qz, dz) * B1(qy, dy) * B2(qx, dx)
     //                   - f(0, qz, qy, qx) * G2(qz, dz) * B1(qy, dy) * B2(qx, dx)
     //                   + f(2, qz, qy, qx) * B2(qz, dz) * B1(qy, dy) * G2(qx, dx);
-    cache.A2.y[0] = contract< x, 0 >(source[1], B2) + contract< x, 0 >(flux[2], G2);
-    cache.A2.y[1] = contract< x, 0 >(flux[0], B2);
-    cache.A1.y = contract< z, 0 >(cache.A2.y[0], B2) - contract< z, 0 >(cache.A2.y[1], G2);
-    element_residual[0].y += contract< y, 0 >(cache.A1.y, B1);
+    {
+      auto A20 = contract< x, 0 >(source[1], B2) + contract< x, 0 >(flux[2], G2);
+      auto A21 = contract< x, 0 >(flux[0], B2);
+      auto A1 = contract< z, 0 >(A20, B2) - contract< z, 0 >(A21, G2);
+      element_residual[0].y += contract< y, 0 >(A1, B1);
+    }
 
     //  r(2, dz, dy, dx) = s(2, qz, qy, qx) * B1(qz, dz) * B2(qy, dy) * B2(qx, dx) 
     //                   + f(0, qz, qy, qx) * B1(qz, dz) * G2(qy, dy) * B2(qx, dx) 
     //                   - f(1, qz, qy, qx) * B1(qz, dz) * B2(qy, dy) * G2(qx, dx);
-    cache.A2.z[0] = contract< y, 0 >(source[2], B2) + contract< y, 0 >(flux[0], G2);
-    cache.A2.z[1] = contract< y, 0 >(flux[1], B2);
-    cache.A1.z = contract< x, 0 >(cache.A2.z[0], B2) - contract< x, 0 >(cache.A2.z[1], G2);
-    element_residual[0].z += contract< z, 0 >(cache.A1.z, B1);
+    {
+      auto A20 = contract< y, 0 >(source[2], B2) + contract< y, 0 >(flux[0], G2);
+      auto A21 = contract< y, 0 >(flux[1], B2);
+      auto A1 = contract< x, 0 >(A20, B2) - contract< x, 0 >(A21, G2);
+      element_residual[0].z += contract< z, 0 >(A1, B1);
+    }
     // clang-format on
   }
 
   //////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#if defined(__CUDACC__)
+#if 0

```

Comment:
Is this disabled for now because we can't dynamically allocate the temporary tensors on device? What is the plan for this moving forward?

Reply:
it's not really disabling it now, the code wasn't being used before. 

> What is the plan for this moving forward?

When I refactor these kernels to use runtime-configurable polynomial orders, I'd like to get everything up and running on CUDA again (but there are still some additional changes that need to be made to the physics modules to work with GPUs).

Reply:
I would add a comment to reflect this, but that makes sense to me.

Reply:
noted here: https://github.com/LLNL/serac/issues/881